server/controller/src/main/java/ai/starwhale/mlops/domain/job/JobService.java

```diff
@@ -185,9 +185,10 @@ public Long createJob(String projectUrl,
                 .map(datasetService::findDatasetVersion)
                 .collect(Collectors.toMap(DatasetVersion::getId, DatasetVersion::getVersionName));
 
-        if (!StringUtils.hasText(stepSpecOverWrites) && !StringUtils.hasText(handler)) {
+        if ((!StringUtils.hasText(stepSpecOverWrites) && !StringUtils.hasText(handler))
+                || StringUtils.hasText(stepSpecOverWrites) && StringUtils.hasText(handler)) {
```

Comment:
It will be clearer to wrap the last two items in parentheses.

Reply:
fixed